### ISSUE TITLE
Other potential Error source for "Error: Cannot find symbol for node: x"

### ISSUE THREAD

**Issue by Uzlopak** [timocov/dts-bundle-generator#173]
Hi, 

first of all: thank for this great product. 

tldr:
Instead of using named imports use dynamic imports when hitting the "Cannot find symbol for node"-Error.

long version:
I use it very intensive to create my definition files. I use it to create base packages, which get consumed by other packages, which get then get consumed by my projects.

When generating one of my base packages, which get consumed by another package, i got "Error: Cannot find symbol for node: amqplib". I tried everything to figure out why it was erroring. installed everywhere @types/amqplib but could not solve the issue. 

Then i set in tsconfig skipLibCheck to false and got a different error:

```
node_modules/project/amqp/types/index.d.ts(29,72): error TS2503: Cannot find namespace 'amqplib'.
node_modules/project/amqp/types/index.d.ts(49,20): error TS2304: Cannot find name 'Channel'.
node_modules/project/amqp/types/index.d.ts(69,53): error TS2503: Cannot find namespace 'amqplib'.
node_modules/project/amqp/types/index.d.ts(71,21): error TS2503: Cannot find namespace 'amqplib'.
node_modules/project/amqp/types/index.d.ts(77,46): error TS2503: Cannot find namespace 'amqplib'.
node_modules/project/amqp/types/index.d.ts(78,52): error TS2503: Cannot find namespace 'amqplib'.
```
Channel is an interface from amqplib.

I was importing it by doing

```typescript
import { Channel } from amqplib;
```
And was referenced by `Channel`.
So I changed the named import to a dynamic import.

```typescript
import * as amqplib from amqplib;
```

And referenced it as `amqplib.Channel`.

And then dts-bundle-generator worked as expected.

Seems like dts-bundle-generator does not like named imports for types. Or maybe it fails in tsc. Not important where ;).

I dont know if this was already explicitly documented, but I strongly recommend to document it. Took some time to figure it out by myself, and all the other search results were not hinting to this problem. So if somebody hits the same issue, he/she should first check for potential named imports and change them to dynamic imports.

**Comment by timocov** [comment]
Hi @Uzlopak, do you have an example where I can reproduce the issue?

**Comment by Uzlopak** [comment]
I tried to create a mvce by using npm link but the issue does not show up.

For my project I publish into a private artifact server. As such maybe fails to realize to use the @types/amqplib of the $actualPackage, which uses that $peerPackage and tries to look for some reason into the node_modules of the $peerPackage and as @types/amqplib is not in node_modules/$peerDependency/node_modules  when I pull them from my artifact server (node_modules is not in the $peerDependency and @types/amqplib is a devDependency in the $peerDependency) fails to find the correct @types/amqplib in the $actualPackage i want to build the types for. Maybe thats why i cant reproduce it with npm link, as the folder gets just symlinked with node_modules, etc.

I can understand if you want to close this issue. I just wanted to document this behaviour and hope that google indexes this page and if somebody searches for a similar issue finds this and can check if this is also an issue for him/her. Also this is why I dont recommend always to do dynamic imports (I like named imports better tbh), but only if you maybe hit this issue and want to exclude that the named import is the reason for the error.

:)

**Comment by timocov** [comment]
>  I pull them from my artifact server (node_modules is not in the published $peerDependency and @types/amqplib is a devDependency in the $peerDependency and not downloaded when installing the $peerPackage) fails to find the correct @types/amqplib in the $actualPackage i want to build the types for. Maybe thats why i cant reproduce it with npm link, as the folder gets just symlinked with node_modules, etc.

@Uzlopak You can pack your package to tar.gz via `npm pack` command and then install it as a package (just a path to your tar.gz file in package.json), [see example here](https://github.com/puppeteer/puppeteer/blob/ad7f1de44f78df4a304c8e052935e53d4685530c/test-ts-types/js-cjs-import-cjs-output/package.json#L10).

> I can understand if you want to close this issue. I just wanted to document this behaviour and hope that google indexes this page and if somebody searches for a similar issue finds this and can check if this is also an issue for him/her. Also this is why I dont recommend always to do dynamic imports (I like named imports better tbh), but only if you maybe hit this issue and want to exclude that the named import is the reason for the error.

To document something I need to understand it first. Right now I have no idea what is going on because I didn't see the reproduction of the issue 🙂

**Comment by timocov** [comment]
@Uzlopak do you have any news?

**Comment by Uzlopak** [comment]
Hi timocov, 

I have to admin: No. I learn from this issue, that I should encapsulate my plugins better, so that I wont need any external typings anymore. 

I close this issue, as it was a result of bad practice anyway.

Thank you for your time! Thanks again for the great product!
